moderation.py

```diff
@@ -0,0 +1,114 @@
+import asyncio
+import re
+
+import discord
+from discord.ext import commands
+
+
+class HierarchyMember(commands.MemberConverter):
+    """
+    A member converter that respects Discords role hierarchy system.
+    """
+    async def convert(self, ctx, arg):
+        member = await super().convert(ctx, arg)
+
+        can_do_action = (ctx.author == ctx.guild.owner or
+                         ctx.author.top_role > member.top_role)
+        if not can_do_action:
+            raise commands.BadArgument("Role hierarchy prevents you from doing this.")
+        return member
+
+
+class HackMember(HierarchyMember):
+    """
+    A member converter that also allows arbitrary IDs to be passed.
+    """
+    async def convert(self, ctx, arg):
+        try:
+            member = await super().convert(ctx, arg)
+            return member.id
+        except commands.BadArgument:
+            try:
+                return int(arg)
+            except ValueError:
+                raise commands.BadArgument(f'"{arg}" is not a valid member or ID.')
+
+
+class BannedUser(commands.Converter):
+    """
+    A converter for members who have been banned.
+    """
+    async def convert(self, ctx, arg):
+        bans = [entry.user for entry in await ctx.guild.bans()]
+
+        try:
+            user = discord.utils.get(bans, id=int(arg))
```

Comment:
Why is this flattened and then filtered? Why not loop over and break if you hit the user?